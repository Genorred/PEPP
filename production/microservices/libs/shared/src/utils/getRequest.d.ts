import { ExecutionContext } from "@nestjs/common";
import { CustomContext } from "@_shared/types/CustomContext";
declare function getRequest(context: ExecutionContext): CustomContext["req"];
export default getRequest;
