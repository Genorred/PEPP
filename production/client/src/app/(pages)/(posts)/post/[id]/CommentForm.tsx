import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
import { Textarea } from "@/shared/ui/textarea";
import { Button } from "@/shared/ui/button";
import { useSelector } from "react-redux";
import { userSlice } from "@/entities/User/model/user.slice";
import Link from "next/link";

const CommentForm = ({ isReplyingState, onCreate, placeholder = "comment" }: {
  isReplyingState?: [boolean, Dispatch<SetStateAction<boolean>>];
  onCreate: (message: string, onError?: () => void) => void;
  placeholder?: string
}) => {
  const [isWriting, setIsWriting] = isReplyingState ?? useState(false);
  const [replyMessage, setReplyMessage] = useState("");
  const [sentReplyMessage, setSentReplyMessage] = useState("");
  const user = useSelector(userSlice.selectors.user);
  useEffect(() => {
    if (!user)
      setIsWriting(false);
  }, [user]);


  function onSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    if (replyMessage.length > 0) {
      setSentReplyMessage(replyMessage);
      onCreate(replyMessage.trim(), () => {
        setReplyMessage(sentReplyMessage)
      });
      setReplyMessage('')
    }
  }

  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setReplyMessage(e.target.value);
  };

  return (
    <>
      <form className="mt-4" onSubmit={onSubmit}>
        <Textarea
          disabled={!user}
          onFocus={() => setIsWriting(true)}
          onChange={onChange}
          value={replyMessage}
          placeholder={`Write your ${placeholder}...`}
          className="min-h-[100px] mb-2"
        />
        {isWriting && (
          <div className="mt-4 flex justify-end space-x-2">
            <Button variant="outline" type={"button"} size="sm" onClick={() => setIsWriting(false)}>
              Cancel
            </Button>
            <Button size="sm" disabled={replyMessage.length < 0}>
              Submit Message
            </Button>
          </div>
        )}
        {!user ?
          <Link href={"/sign-in"}>Authorize</Link>
          : null
        }
      </form>

    </>
  );
};

export default CommentForm;