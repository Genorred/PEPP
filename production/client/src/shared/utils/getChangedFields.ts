import { isEqual, omitBy } from "lodash";

export function getChangedFields<T extends Record<any, any>>(original: T, updated: T) {
  return omitBy(updated, (value, key) => isEqual(value, original[key]));
}