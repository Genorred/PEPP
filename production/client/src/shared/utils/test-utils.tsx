import React, { PropsWithChildren } from "react";
import type { RenderOptions } from "@testing-library/react";
import { render } from "@testing-library/react";
import { configureStore, createSlice, Slice } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { rootReducer } from "@/shared/lib/redux";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import { queryClient } from "@/shared/api/queryClient";

const anySlice = createSlice({
  name: "any",
  initialState: {},
  selectors: {},
  reducers: {}
}).injectInto(rootReducer);

interface ExtendedRenderOptions<SlicesT extends [Slice]> extends Omit<RenderOptions, "queries"> {
  slices?: SlicesT;
  preloadedState?: {
    [SliceItem in SlicesT[number] as SliceItem["name"]]: ReturnType<SliceItem["getInitialState"]>
  };
}

export function setupStore(preloadedState?: Record<string, any>) {
  return configureStore({
    reducer: rootReducer,
    preloadedState
  });
}

export function renderWithProviders<SlicesT extends [Slice]>(
  ui: React.ReactElement,
  extendedRenderOptions: ExtendedRenderOptions<SlicesT> = {}
) {
  const {
    preloadedState = {},
    ...renderOptions
  } = extendedRenderOptions;

  const store = setupStore(preloadedState);
  const queryClient = new QueryClient({
    defaultOptions: {

    }
  });

  const Wrapper = ({ children }: PropsWithChildren) => {
    return (
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>{children}</Provider>
      </QueryClientProvider>
    );
  };


  return {
    store,
    wrapper: Wrapper,
    ...render(ui, { wrapper: Wrapper, ...renderOptions })
  };
}