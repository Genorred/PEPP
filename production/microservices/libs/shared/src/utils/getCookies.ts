import { Credentials, CustomContext } from "@_shared/types/CustomContext";

const getCookies = (context: CustomContext) => JSON.parse(context.req.headers.cookies) as Credentials | undefined;
export default getCookies;