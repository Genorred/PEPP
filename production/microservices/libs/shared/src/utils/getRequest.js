"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const graphql_1 = require("@nestjs/graphql");

function getRequest(context) {
  const ctx = graphql_1.GqlExecutionContext.create(context);
  console.log("rekvest", ctx.getContext());
  return ctx.getContext().req;
}

exports.default = getRequest;
//# sourceMappingURL=getRequest.js.map