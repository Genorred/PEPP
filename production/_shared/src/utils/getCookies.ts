import { ExecutionContext } from "@nestjs/common";
import getRequest from "./getRequest";
import { JwtPayload } from "@_shared/entities/jwt.entity";
import { Credentials, CustomContext } from "@_shared/types/CustomContext";

const getCookies = (context: CustomContext) => JSON.parse(context.req.headers.cookies) as Credentials | undefined;
export default getCookies;