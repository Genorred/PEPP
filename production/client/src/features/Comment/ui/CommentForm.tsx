"use client";
import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
import { Textarea } from "@/shared/ui/textarea";
import { Button, buttonVariants } from "@/shared/ui/button";
import { useSelector } from "react-redux";
import { userSlice } from "@/entities/User/model/user.slice";
import Link from "next/link";
import { cn } from "@/shared/lib/utils";
import { usePathname } from "next/navigation";
import { useAppStore } from "@/shared/lib/redux";

const CommentForm = ({ isReplyingState, onCreate, placeholder = "comment" }: {
  isReplyingState?: [boolean, Dispatch<SetStateAction<boolean>>];
  onCreate: (message: string, onError?: () => void) => void;
  placeholder?: string
}) => {
  const [isWriting, setIsWriting] = isReplyingState ?? useState(false);
  const [replyMessage, setReplyMessage] = useState("");
  const [sentReplyMessage, setSentReplyMessage] = useState("");
  const user = useSelector(userSlice.selectors.user);
  useEffect(() => {
    if (!user)
      setIsWriting(false);
  }, [user]);


  function onSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    if (replyMessage.length > 0) {
      setSentReplyMessage(replyMessage);
      onCreate(replyMessage.trim(), () => {
        setReplyMessage(sentReplyMessage);
      });
      setReplyMessage("");
    }
  }

  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setReplyMessage(e.target.value);
  };
  const url = usePathname();

  console.log(user);
  console.log(userSlice.selectors.user.unwrapped.toString());
  console.log('render disabled', !user);
  const onFocus = () => {
    if (user) setIsWriting(true)
  }
  return (
    <>
      <form className="mt-4" onSubmit={onSubmit}>
        <Textarea
          data-testid="comment-form"
          disabled={!user}
          onFocus={onFocus}
          onChange={onChange}
          value={replyMessage}
          placeholder={`Write your ${placeholder}...`}
          className="min-h-[100px] mb-2"
        />
        {isWriting && (
          <div className="mt-4 flex justify-end space-x-2">
            <Button variant="outline" type={"button"} size="sm" onClick={() => setIsWriting(false)}
                    data-testid="comment-form-button-cancel">
              Cancel
            </Button>
            <Button size="sm" disabled={replyMessage.length < 0} data-testid="comment-form-button-submit">
              Submit Message
            </Button>
          </div>
        )}
        {!user ?
          <Link href={`/sign-in?returnUrl=${url}`} className={cn(buttonVariants({
            variant: "default"
          }), "ml-auto")}>Authorize</Link>
          : null
        }
      </form>

    </>
  );
};

export default CommentForm;