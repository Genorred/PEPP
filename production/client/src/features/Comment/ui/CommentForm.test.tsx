import { fireEvent, screen } from "@testing-library/react";
import CommentForm from "./CommentForm";
import { renderWithProviders } from "@/shared/utils/test-utils";
import { userSlice } from "@/entities/User/model/user.slice";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

const createFunction = jest.fn();
const userInitialState = {
  id: 1,
  username: "j",
  email: "",
  createdAt: "",
  expireDate: 999999
};
describe("Comment form", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should open on focus", () => {
    const { asFragment } = renderWithProviders(<CommentForm onCreate={createFunction} />, {
      slices: [userSlice],
      preloadedState: {
        user: userInitialState
      }
    });

    const textarea = screen.getByTestId("comment-form");
    expect(screen.queryByTestId("comment-form-button-submit")).not.toBeInTheDocument();
    fireEvent.focus(textarea);
    expect(screen.getByTestId("comment-form-button-submit")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
  it("should close on cancel button", async () => {
    const { asFragment } = renderWithProviders(<CommentForm onCreate={createFunction} />, {
      slices: [userSlice],
      preloadedState: {
        user: userInitialState
      }
    });

    const textarea = screen.getByTestId("comment-form");
    expect(screen.queryByTestId("comment-form-button-cancel")).not.toBeInTheDocument();

    fireEvent.focus(textarea);
    const cancelButton = screen.getByTestId("comment-form-button-cancel");
    expect(cancelButton).toBeInTheDocument();

    await userEvent.click(cancelButton);
    expect(screen.queryByTestId("comment-form-button-cancel")).toBeNull();
    expect(asFragment).toMatchSnapshot();
  });
  it("should send message on submit", async () => {
    const { asFragment } = renderWithProviders(<CommentForm onCreate={createFunction} />, {
      slices: [userSlice],
      preloadedState: {
        user: userInitialState
      }
    });
    const textarea = screen.getByTestId("comment-form");
    fireEvent.focus(textarea);
    const message = "Satisfying";
    fireEvent.input(textarea, {
      target: {
        value: message
      }
    });
    expect(asFragment()).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("comment-form-button-submit"));
    expect(createFunction).toHaveBeenCalledTimes(1);
    expect(createFunction).toHaveBeenCalledWith(message, expect.anything());
  });
});