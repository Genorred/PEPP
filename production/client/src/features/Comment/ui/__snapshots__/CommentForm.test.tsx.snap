// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment form should close on cancel button 1`] = `[Function]`;

exports[`Comment form should open on focus 1`] = `
<DocumentFragment>
  <form
    class="mt-4"
  >
    <textarea
      class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px] mb-2"
      data-testid="comment-form"
      disabled=""
      placeholder="Write your comment..."
    />
    <div
      class="mt-4 flex justify-end space-x-2"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
        data-testid="comment-form-button-cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3"
        data-testid="comment-form-button-submit"
      >
        Submit Message
      </button>
    </div>
    <a
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 ml-auto"
      href="/sign-in?returnUrl=null"
    >
      Authorize
    </a>
  </form>
</DocumentFragment>
`;

exports[`Comment form should send message on submit 1`] = `
<DocumentFragment>
  <form
    class="mt-4"
  >
    <textarea
      class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px] mb-2"
      data-testid="comment-form"
      disabled=""
      placeholder="Write your comment..."
    >
      Satisfying
    </textarea>
    <div
      class="mt-4 flex justify-end space-x-2"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
        data-testid="comment-form-button-cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3"
        data-testid="comment-form-button-submit"
      >
        Submit Message
      </button>
    </div>
    <a
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 ml-auto"
      href="/sign-in?returnUrl=null"
    >
      Authorize
    </a>
  </form>
</DocumentFragment>
`;
